apps/snfoundry/contracts/src/marketplace.cairo

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: MIT
+// Compatible with OpenZeppelin Contracts for Cairo ^0.18.0
+
+#[starknet::contract]
+mod Marketplace {
+    // dispatchers
+    use contracts::cofi_collection::{ICofiCollectionDispatcher, ICofiCollectionDispatcherTrait};
+    use openzeppelin::access::accesscontrol::AccessControlComponent;
+    use openzeppelin::access::accesscontrol::DEFAULT_ADMIN_ROLE;
```

Comment:
Not sure how many roles should we have for this contract... is there buyer and seller roles? or anyone can do that?